### DISCUSSION TITLE
ARM suddenly "fatal error"s every rip when calling MKV

### DISCUSSION THREAD

**Discussion by mdalen1** [automatic-ripping-machine/automatic-ripping-machine#850] [Q&A] [answered]
I have been using this wonderful piece of software for a while, after someone gave me a truckload of DVD's. It has worked fine for around 2 500 DVDs the last few months. When I started up a new pile today after no ripping for a week, all DVDs now fail at the same point. It seems to be when MKV is started. I have made no hardware or software changes to the machine. It is running in docker on top of Debian minimal install. After the issue arose I ran apt-get upgrade, without any improvement. I have five DVD readers, all worked fine before and all have the same error now. I have tested with 9 different DVDs a total of 16 times. Data dvd's rip fine, it seems this is just related to DVD's, so MakeMKV. What I assume is the relevant part of the log is copied in below. Any suggestions on how to fix this?

[02-06-2023 11:03:57] INFO ARM: ************* Ripping disc with MakeMKV *************
[02-06-2023 11:03:57] INFO ARM: Updating MakeMKV key...
MakeMKV beta key for this month: T-LrMaMHZQshJkxCXaUIhAlwmQW3aFFsgWdR1MbMd43RUtQG8Qhijh2FEOBa5BjpQkv4
/home/arm/.MakeMKV/settings.conf exists, updating value of app_Key
[02-06-2023 11:03:58] INFO ARM: Starting MakeMKV rip. Method is mkv
[02-06-2023 11:03:58] ERROR ARM: Call to MakeMKV failed with code: 1 (b'')
[02-06-2023 11:03:58] ERROR ARM: MakeMKV did not complete successfully.  Exiting ARM! Error: exceptions must derive from BaseException
[02-06-2023 11:03:58] ERROR ARM: A fatal error has occurred and ARM is exiting.  See traceback below for details.

**Comment by 1337-server** [comment]
Can you re-run in debug mode ?

**Comment by mdalen1** [comment]
[log.txt](https://github.com/automatic-ripping-machine/automatic-ripping-machine/files/11635272/log.txt)
Re-ran in log debug mode, full log attached.

**Comment by 1337-server** [comment, marked as answer]
Could you attach to the container and run `makemkvcon -r info`

**Comment by mdalen1** [comment]
Thanks for the help, seems like this found the problem. Makemkv complains I am using a version that is too old for the key. apt-get does not find makemkv, not sure how I should go about upgrading without breaking anything?

root@feb810f10aa3:/home/arm# makemkvcon -r info
MSG:1005,0,1,"MakeMKV v1.17.3 linux(x64-release) started","%1 started","MakeMKV v1.17.3 linux(x64-release)"
MSG:3338,0,2,"Downloading latest SDF to /root/.MakeMKV ...","Downloading latest %1 to %2 ...","SDF","/root/.MakeMKV"
DRV:0,0,999,0,"DVD+R-DL TSSTcorp CD/DVDW TS-H653L HA03 TS-H653LFirmware","","/dev/sr1"
DRV:1,0,999,0,"DVD+R-DL ASUS SH-224DB 1.10 S11E68CF5009FY","","/dev/sr4"
DRV:2,0,999,0,"DVD+R-DL hp DVD A  DH16AESH 3HH5 2B8352906125","","/dev/sr3"
DRV:3,0,999,0,"DVD+R-DL TSSTcorp CDDVDW SH-224BB SB00 R8WS6GBD200JPS","","/dev/sr0"
DRV:4,0,999,0,"DVD+R-DL hp DVD A  DH16AESH 3HH5 2C8352905316","","/dev/sr2"
DRV:5,256,999,0,"","",""
DRV:6,256,999,0,"","",""
DRV:7,256,999,0,"","",""
DRV:8,256,999,0,"","",""
DRV:9,256,999,0,"","",""
DRV:10,256,999,0,"","",""
DRV:11,256,999,0,"","",""
DRV:12,256,999,0,"","",""
DRV:13,256,999,0,"","",""
DRV:14,256,999,0,"","",""
DRV:15,256,999,0,"","",""
MSG:5021,131332,1,"This application version is too old.  Please download the latest version at http://www.makemkv.com/ or enter a registration key to continue using the current version.","This application version is too old.  Please download the latest version at %1 or enter a registration key to continue using the current version.","http://www.makemkv.com/"

**Comment by mdalen1** [comment]
I tried updating makemkv to 1-17-4 inside the container with the instructions here: [[,](https://forum.makemkv.com/forum/viewtopic.php?f=3&t=224)](https://forum.makemkv.com/forum/viewtopic.php?f=3&t=224)  , but makemkvcon -r info still reads 1.17.3.

**Comment by mdalen1** [comment]
Fixed it! I copied the new makemkv* executables from /usr/bin to /usr/local/bin . I am sure there is a more correct way to fix it, but this works. :-)
Thanks for the hints!
